### ISSUE TITLE
Research impact of techniques in the WCAG 1.4.4. Guideline for resizing text 

### ISSUE THREAD

**Issue by roslynwythe** [hackforla/website#4227]
### Dependency
- [ ] After #1347 is closed, review this issue to see if it needs rewriting or closing.  See comments 

### Overview
Research the feasibility and impact of implementing alternative Accessibility techniques, so that Hack for LA's website can score the highest Accessibility rating possible, while still being easy to use for all visitors.

### Details
Zoom functionality in modern browsers ensures that the WCAG 1.4.4 guideline on resizing text is satisfied, but there are alternative techniques for compliance that offer accessibility advantages.  In this issue we will research the feasibility and impact of implementing alternative techniques described in the guideline, in particular:  (1) providing accessibility controls on the page, and (2) Specifying relative font sizes and resizing text containers and layout regions to scale with text size

### Action Items

- [ ] Research the impact of implementing:
   - [G178: Providing controls on the Web page that allow users to incrementally change the size of all text on the page up to 200 percent](https://www.w3.org/WAI/WCAG21/Techniques/general/G178.html) 
- [ ] Research the alternative of ensuring that text containers resize when the text resizes AND using measurements that are relative to other measurements in the content by using one or more of the following techniques:
  - [C28: Specifying the size of text containers using em units](https://www.w3.org/WAI/WCAG21/Techniques/css/C28.html)
  - Techniques for relative measurements
     - [C12: Using percent for font sizes](https://www.w3.org/WAI/WCAG21/Techniques/css/C12.html)
     - [C13: Using named font sizes](https://www.w3.org/WAI/WCAG21/Techniques/css/C13.html)
     - [C14: Using em units for font sizes](https://www.w3.org/WAI/WCAG21/Techniques/css/C14.html)
  - Techniques for text container resizing
     - [SCR34: Calculating size and position in a way that scales with text size](https://www.w3.org/WAI/WCAG21/Techniques/client-side-script/SCR34.html)
    - [G146: Using liquid layout](https://www.w3.org/WAI/WCAG21/Techniques/general/G146.html)
- [ ] Document findings in a Decision Record.  

### Resources/Instructions
- [W3C Web Content Accessibility Guidelines (WCAG) 2.0](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#visual-audio-contrast)
- [Understanding Success Criteria 1.4.4: Resize Text](https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html)
- [Sufficient techniques for WCAG 1.4.4](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0&showtechniques=144#qr-visual-audio-contrast-scale)
- [How to change text size in Google Chrome](https://support.google.com/chrome/answer/96810?hl=en&co=GENIE.Platform%3DDesktop)

### Created by this issue
- DRAFT: [Decision Record - Hack for LA Issue #4227](https://docs.google.com/document/d/1CneoiEo4GDFXXWJnaZH56XyYUBGSKnlG2ZiP8L24eiE/edit)

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe what complexity is this?

**Comment by roslynwythe** [comment]
@ExperimentsInHonesty  To determine the complexity, I would appreciate your input on the scope of this issue.  Should this issue focus just on the evaluation of compliance or should it include analysis/decision re: alternative CSS approaches?  And should we broaden the focus to include not just fixed font size but also other CSS issues that could impact text scaling, such as fixed width containers or the use of text within images?

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe good questions.  i think I will need a better understanding of the type of rabbit hole we are looking at.  The last accessibility audit we did really was focused on alt and not using h1-5 tags correctly (using some standard testing tools specified in this article [Which accessibility testing tool should you use?](https://medium.com/pulsar/which-accessibility-testing-tool-should-you-use-e5990e6ef0a).  So I would love to know what if any testing tools we would be using for this round and just generally more about these new requirements.

**Comment by roslynwythe** [comment]
@ExperimentsInHonesty Thank you for the link to the article [Which accessibility testing tool should you use?](https://medium.com/pulsar/which-accessibility-testing-tool-should-you-use-e5990e6ef0a)  I ran a few pages against two accessibility checkers and our pages did not fail SC 1.4.4.  

Use of absolute units is not a recommended practice, because it overrides the user's style sheet.  You can see this by going to your chrome Appearance settings and changing the font size to Large.  On our site there will be no change.  However, unless a site specifically restricts it, all modern browsers have a built-in zoom function, and that is considered an alternate means of scaling text, thus allowing SC 1.4.4 to be fulfilled.  

Other accessibility issues were identified and issues will be ready shortly

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe Thanks for the update.  Sounds like this issue is ready for a decision record.

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe When making the new issues, just confirm that we don't duplicate the items that are still open from that audit.  Those items can be found by looking at issues with the [label feature: accessibility](https://github.com/hackforla/website/projects/7?card_filter_query=label%3A%22feature%3A+accessibility%22)

**Comment by ExperimentsInHonesty** [comment]
p.s. please assign size and complexity labels to this issue

**Comment by roslynwythe** [comment]
@ExperimentsInHonesty This issue has been completely rewritten.  It now explores the impact of implementing several techniques which would make it easier for users to resize the text on the website.

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe at the top of this issue in the `overview` section the text says:

> layour regions

I think you meant layout regions.


Other than that, this looks good.  If you are ready for me to prioritize it, add the `ready for prioritization` label.

**Comment by kevin31yu** [comment]
i. Availability: Monday-Friday (7-9)
ii. ETA: 11/22-11/24

**Comment by kevin31yu** [comment]
My apologies, my schedule was changed and I am available after 11/29.

**Comment by kevin31yu** [comment]
Progress: Completed reading and checked off the tasks
Blockers: I am not sure what to next
Availability: 8 hrs
ETA: Not sure

**Comment by kevin31yu** [comment]
I have unchecked all the boxes for the next assignee to work with.

**Comment by Anahisv23** [comment]
i. Availability: Wednesday Feb 7 from 11am - 4pm
ii. ETA: Friday Feb 9

**Comment by Anahisv23** [comment]
Hi, below I have attached a google doc with a decision record including my findings of the impact of implementing G178 and alternative techniques C28, C12, C13, C14, SCR34, and G146 to comply with the WCAG 1.4.4 guideline on resizing text. I appreciate any feedback! 

https://docs.google.com/document/d/1CneoiEo4GDFXXWJnaZH56XyYUBGSKnlG2ZiP8L24eiE/edit?usp=sharing

**Comment by roslynwythe** [comment]
Thank you @Anahisv23.  You have done a lot of great research and prepared a well written summary.  It seems that we have three decisions to make:

1. Do we adopt relative units for font sizes, to accomodate user resized text?
2. If we adopt 1 above, how do we scale containers to accomodate the resized text?
3. Do we provide controls on our web pages to provide a GUI for users to resize text, rather than having to use another method (ie,  browser or system settings or a user style sheet)?

Let's focus on 1 and 2 above. For information about Decision Records, please see 
https://github.com/hackforla/website/wiki/Decision-Records-on-Solutions-Not-Implemented
https://github.com/hackforla/website/wiki/Decision-Records-on-Solutions-Adopted

I do have some questions:

Regarding decision 1:
-  Do you recommend the use of percent font size over the use of em for font size?  If so, please explain why.

Regarding decision 2:
- I was a bit surprised to read that the method of calculating container sizes is considered easier than relative em units.  Did I understant that correctly?  Do you know which method is more commonly in use and why?
- Is the "liquid" layout a separate technique or is it also the use of relative units for text and containers?
- What are the pros and cons of specifying container size as percent vs ems?

**Comment by roslynwythe** [comment]
@ExperimentsInHonesty I think I tried to put too much into this issue.  I believe that before we consider providing accessibility controls on the page, we first need to determine feasibility and techniques for allowing text and containers to resize.  So I suggest removing the first Action Item.  Also I suggest removing the last item, the Decision Record, because that decision should probably be made and documented by a dev lead.  

Also I'm not sure if I should hide the comments with Anahis's work and my partial review

**Comment by ExperimentsInHonesty** [comment]
@Anahisv23 Good job with the analysis. Please move this document into the team Google Drive.

@roslynwythe if I just read the analysis document.  I think we need to talk about this in the merge team meeting and I will add it to the agenda.

**Comment by ExperimentsInHonesty** [comment]
- Add the dependency of the issue that re runs the accessability audit #1347
- add text about rewriting the issue if its still needed after the audit.  See comment above.

**Comment by Anahisv23** [comment]
Hi @ExperimentsInHonesty, thanks! I want to add the document into the team google drive but I don't currently have access so I requested access. As soon as I get access I will move the document there.

**Comment by roslynwythe** [comment]
@Anahisv23 you have contributor rights on the "HackforLA.org website" share, In that share, please place the file in the folder "HackforLA.org website folder/Files related to dev issues/issue 4227  Thanks!

**Comment by Anahisv23** [comment]
Hey @roslynwythe is there a link I can access to join the google drive bcs the link I'm currently using states I do not have access? I'll include the link below. 

https://drive.google.com/drive/u/0/folders/1p76K0FgfiAWeIIEyoyJ_Iik8FVj8cBjT

**Comment by ExperimentsInHonesty** [comment]
@Anahisv23 I added you to the drive using the email address from the hackforla website roster.

**Comment by Anahisv23** [comment]
Thanks @ExperimentsInHonesty and @roslynwythe. I have added the document to the google drive.

**Comment by ExperimentsInHonesty** [comment]
@Anahisv23 Thanks for the update.  @roslynwythe please review Google Doc: [Decision Record - Hack for LA Issue #4227](https://docs.google.com/document/d/1CneoiEo4GDFXXWJnaZH56XyYUBGSKnlG2ZiP8L24eiE/edit)

**Comment by ExperimentsInHonesty** [comment]
@Anahisv23 
I went and looked at this link: [Decision Record - Hack for LA Issue #4227](https://docs.google.com/document/d/1CneoiEo4GDFXXWJnaZH56XyYUBGSKnlG2ZiP8L24eiE/edit) and its a google doc, but its not in the website project drive.  You have made it so that anyone can edit the file, but that is not the same as moving it to our google drive.  Additionally, there is a pdf in that folder, but we need the editable Google doc

- Please place the file in the folder [HackforLA.org website folder/3 - Developers/Files related to dev issues/issue 4227](https://drive.google.com/drive/folders/1DHoW1znCXYL9jhQLjXjVPckgwPfYLVhE)

**Comment by Anahisv23** [comment]
@ExperimentsInHonesty Hi Bonnie, I took a look at the drive and you are right. I had only included a pdf of the google doc. I went ahead and added the actual google doc into the folder HackforLA.org website folder/3 - Developers/Files related to dev issues/issue 4227. Let me know if you have any trouble accessing it. 

I also noticed I've been assigned to this issue again, even though I had unassigned myself back in February. Could you provide some context on why I've been reassigned?

**Comment by Anahisv23** [comment]
Hi @ExperimentsInHonesty hope you're well! I addressed the google drive issues that you mentioned. I went ahead an un-assigned myself from this issue and placed it back on the prioritized backlog for another dev to work on. I spoke to Roslyn about un-assigning myself from the issue back in February to confirm my un-assignment.
